### PR TITLE
Fix #3891: Add challenge `domain`

### LINKED ISSUE #3891: Add challenge domains to challenges
### Description
Currently it is difficult to find challenges by domain (say CV, NLP, RL), etc. This issue is a first step towards that direction where we add the domain a particular challenge belongs to and display it on the challenge page in this section:

Screenshot 2023-03-20 at 12 51 31 PM

The challenge domain must be made a part of the challenge configuration and the [EvalAI-Starters](https://github.com/Cloud-CV/EvalAI-Starters) repository.
This will also require you to add changes to the [challenge validation script](https://github.com/Cloud-CV/EvalAI/blob/19130726790e4cb79b0d2e8603dd6817b16dba69/apps/challenges/challenge_config_utils.py#L223) to ensure people put appropriate values for domains. Please create a PR with these dummies - `[Computer Vision, Natural Language Processing, Reinforcement Learning]`,  before to converge to the set of domains we might want to include.

### Pre-requisites/Learnings
This issue essentially requires background on a few things:

- An understanding of the [challenge configuration](https://evalai.readthedocs.io/en/latest/configuration.html).
- A high-level understanding of [EvalAI challenges](https://evalai.readthedocs.io/en/latest/host_challenge.html).
- An understanding of [our frontend](https://github.com/Cloud-CV/EvalAI/tree/master/frontend).
- How the [challenge validation](https://github.com/Cloud-CV/EvalAI/blob/19130726790e4cb79b0d2e8603dd6817b16dba69/apps/challenges/challenge_config_utils.py#L223) works.

### How to Approach
- You can start by setting up EvalAI locally. Set up the dev version locally (preferred) using the steps [here](https://github.com/Cloud-CV/EvalAI/tree/master#installation-instructions).
- Add a new field to the challenge model in Django.
- Play around with the challenge configuration and the frontend, put up screenshots.
- Update the [challenge validation script](https://github.com/Cloud-CV/EvalAI/blob/19130726790e4cb79b0d2e8603dd6817b16dba69/apps/challenges/challenge_config_utils.py#L223) to check for the newly added field.
- Reach out to us regarding any issues that you face or if you need any help with understanding any of the above.

### PR DESCRIPTION
**Pull request by priyapahwa**
**Preview:**

domain-choice

_This PR includes:_
- Add new challenge model field `domain`
- Modify `challenge_config_utils.py` to check the challenge domain
- Include the corresponding _serializer_ and _test_ fields

**Description:**  The model `Challenge` has been introduced with a new attribute `domain`. The domain of the challenge is considered as a `CharField` type which is limited to 50 characters and must be one of the following options: 'CV' for Computer Vision, 'NLP' for Natural Language Processing, or 'RL' for Reinforcement Learning.

_Fixes:_ #3891

### CODE REVIEW

File: apps/challenges/admin.py [hunk 1]
```diff
@@ -42,6 +42,7 @@ class ChallengeAdmin(ImportExportTimeStampedAdmin):
     list_display = (
         "id",
         "title",
+        "domain",
```
Comment (gchhablani) [inline apps/challenges/admin.py:45]: We do not necessarily need it in the admin page. Please keep this file unchanged.

File: apps/challenges/challenge_config_utils.py [hunk 2]
```diff
@@ -277,6 +277,12 @@ def validate_challenge_config_util(
         message = "Please add the challenge title"
         error_messages.append(message)
 
+    # Check for challenge domain
+    challenge_domain = yaml_file_data.get("domain")
+    if not challenge_domain or challenge_domain not in ["CV", "NLP", "RL"]:
+        message = "Please select a valid challenge domain"
```
Comment (gchhablani) [inline apps/challenges/challenge_config_utils.py:283]: Improve this message to inform that it cannot be empty and it should either of those three values.

File: apps/challenges/serializers.py [hunk 3]
```diff
@@ -237,6 +238,7 @@ class Meta:
         fields = (
             "id",
             "title",
+            "domain",
```
Comment (gchhablani) [inline apps/challenges/serializers.py:241]: We do not necessarily need this. Zip Challenge Serializer is deprecated.

**Review by gchhablani — COMMENTED** [review]
Can you please also provide screenshots from the frontend on your local?

**Comment by priyapahwa** [comment]
Hey @gchhablani 👋🏻 

Thanks for reviewing this PR. I have made the necessary changes.

> Can you please also provide screenshots from the frontend on your local?

Here are the snapshots:

- **Challenge page**

Screenshot from 2023-03-30 00-02-39

- **Challenge list**

Screenshot from 2023-03-29 23-41-18

- **Django Admin**

Screenshot from 2023-03-30 00-03-20

Kindly let me know if this works. Happy to incorporate further modifications as per your suggestions.

**Comment by Suryansh5545** [comment]
@priyapahwa Your build seems to be failing due to the same issue i had in #3904 , you have to add the `"domain": self.challenge.domain` line into every function that is testing this model in test_views.py in challenges and participants folder.

**Comment by priyapahwa** [comment]
> @priyapahwa Your build seems to be failing due to the same issue i had in #3904 , you have to add the `"domain": self.challenge.domain` line into every function that is testing this model in test_views.py in challenges and participants folder.

Hey @Suryansh5545, thanks for your input. 
Yes, I got to know the reason when I checked the build logs. Upon modifying the `test_views.py` file, I am facing some key errors related to `silk` package while running the test suite locally.

**Comment by priyapahwa** [comment]
#3938

**Pull request closed without merging**